pkg/server/server.go

```diff
@@ -222,15 +242,90 @@ func (s *Server) Run(ctx context.Context) error {
 		KeyFile:       s.cfg.EtcdClientInfo.KeyFile,
 		TrustedCAFile: s.cfg.EtcdClientInfo.TrustedCAFile,
 	}
+
+	// TODO(ncdc): I thought I was going to need this, but it turns out this breaks the CRD controllers because they
+	// try to issue Update() calls using the * client, which ends up with the cluster name being set to the default
```

Comment:
Update() with `*` is forbidden ... I have that code in the shardinator but likely it needs to be in kcp proper?